iHome/ihome.frontend/src/app/components/add-device/add-device.component.ts

```diff
@@ -0,0 +1,59 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, Inject, Input, OnInit } from '@angular/core';
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { DeviceToConfigure } from 'src/app/models/device-to-configure';
+import { RoomsApiService } from 'src/app/services/rooms-api.service';
+
+@Component({
+  selector: 'app-add-device',
+  templateUrl: './add-device.component.html',
+  styleUrls: ['./add-device.component.scss']
+})
+export class AddDeviceComponent {
+  @Input() roomId: number = 0;
+  constructor(public dialog: MatDialog) { }
+
+  openDialog() {
+    const dialogRef = this.dialog.open(AddDeviceDialogComponent, {
+      width: '30rem',
+      data: this.roomId
+    });
+    dialogRef.afterClosed().subscribe();
+  }
+
+}
+
+@Component({
+  selector: 'add-device-dialog',
+  templateUrl: './add-device-dialog.component.html',
+  styleUrls: ['./add-device-dialog.component.scss']
+})
+export class AddDeviceDialogComponent implements OnInit{
```

Comment:
Nie wiem czy dwa komponenty w jednym pliku to dobra opcja, jeśli jesteś pewien że nie będzie więcej rzeczy w nich to chyba można zostawić, ale jeśli jest potencjał na rozbudowę ich to ja bym rozdzielił

Reply:
W przykładzie w dokumentacji materiala tak robili dialogi, że mają komponent dialog i oddzielny komponent do uruchamiania dialoga. Kod w jednym pliku też był i tak zostawiłem